aws/tools/cleanup-ssm/src/filter.rs

```diff
@@ -0,0 +1,154 @@
+use crate::TimeProvider;
+use aws_sdk_ssm::types::ParameterMetadata;
+use chrono::{DateTime, Duration};
+
+pub fn filter_old_parameters<T: TimeProvider>(
+    parameters: &[ParameterMetadata],
+    time_provider: &T,
+    older_than_days: u32,
```

Comment:
super minor nit: this could be a much smaller integer type

Reply:
Changed!